src/partinverse.jl

```diff
@@ -0,0 +1,38 @@
+# default methods for when an inverse exists
+export r_inv, l_inv, retraction, coretraction
+
+"""
+    r_inv(function)
```

Comment:
I think we should use a more descriptive name and also be consistent with `inverse`. I suggest
```suggestion
    right_inverse(function)
```